pkg/reconciler/extensions.go

```diff
@@ -0,0 +1,75 @@
+package reconciler
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/operator-framework/helm-operator-plugins/pkg/extension"
+	"helm.sh/helm/v3/pkg/chartutil"
+	"helm.sh/helm/v3/pkg/release"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+)
+
+type extensions struct {
+	extensions []extension.Extension
+}
+
+func (es *extensions) register(e extension.Extension) {
+	es.extensions = append(es.extensions, e)
+}
+
+func (es *extensions) len() int {
+	return len(es.extensions)
+}
+
+func (es *extensions) get(idx int) extension.Extension {
+	return es.extensions[idx]
+}
+
+func (es *extensions) iterate(f func(e extension.Extension) error) error {
```

Comment:
Same question here

Reply:
To reduce code duplication for calling each extension.

Reply:
Makes sense, but I would call it `forEach` or so.

Reply:
ok